packages/core/src/App/Containers/RealAccountSignup/real-account-signup.jsx

```diff
@@ -325,7 +325,9 @@ const RealAccountSignup = ({
                     height={getModalHeight()}
                     width={!has_close_icon ? 'auto' : '904px'}
                 >
-                    <ModalContent state_value={state_value} passthrough={state_index} is_loading={is_loading} />
+                    {is_real_acc_signup_on && (
```

Comment:
How is it possible for `ModalContent` to render if `is_real_acc_signup_on` is `false`?

Reply:
For the first time it works fine. when we want to close Modal we change `is_real_acc_signup_on` and ModalContent. Expected behavior here is that new component in ModalContent not mount. But it mountes and immediately unmount its new children and `promise cancelled` is called here. There is an CSSTransition component in Modal that is causing this problem. CSSTransition's unmount is called after its new children is mounted